packages/dashboard/src/components/layout/MainNavbar/MainNavbar.js

```diff
@@ -1,32 +1,132 @@
-import React from 'react'
+import React, { useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { withRouter } from 'react-router'
 import PropTypes from 'prop-types'
 import classNames from 'classnames'
-import { Container, Navbar } from 'shards-react'
+import { Container, Navbar, Row, Col } from 'shards-react'
+
+import { tradeActions } from '../../../redux/actions'
+import { searchSecurity } from '../../../utils/requests'
 
 import NavbarSearch from './NavbarSearch'
 import NavbarNav from './NavbarNav/NavbarNav'
 import NavbarToggle from './NavbarToggle'
 
-const MainNavbar = ({ layout, stickyTop }) => {
+const DropDownItem = ({
+  name,
+  symbol,
+  exchange,
+  id,
+  currency,
+  selectSecurity,
+}) => {
+  return (
+    <div
+      className="securityDropDownItem"
+      onClick={() => selectSecurity({ id, name, symbol, exchange, currency })}
+    >
+      <Row>
+        <Col className="col-9">{`${name} (${symbol})`}</Col>
+        <Col className="col-3">
+          <i>{exchange}</i>
+        </Col>
+      </Row>
+    </div>
+  )
+}
+
+const MainNavbar = ({ layout, stickyTop, history }) => {
   const classes = classNames(
     'main-navbar',
     'bg-white',
     stickyTop && 'sticky-top'
   )
 
+  const [searchSecurityError, setSearchSecurityError] = useState('')
+  const [securityResults, setSecurityResults] = useState([])
+  const [isSearchLoading, setIsSearchLoading] = useState(false)
+
+  const dispatch = useDispatch()
+
+  const tokens = useSelector(state => state.auth.user.tokens)
+
+  let searchTimer
+
+  const onSearch = e => {
+    clearTimeout(searchTimer)
+    setSearchSecurityError('')
+
+    const query = e.target.value
+    let queryResponse
+
+    // search after user has stopped typing query for 1s
+    searchTimer = setTimeout(async () => {
+      if (!query) {
+        setIsSearchLoading(false)
+        return setSecurityResults([])
+      }
+
+      setIsSearchLoading(true)
+
+      try {
+        queryResponse = (
+          await searchSecurity({
+            tokens: JSON.stringify(tokens),
+            query,
+          })
+        ).data.results
+      } catch (error) {
+        setSearchSecurityError(error.response?.data?.message)
+      }
+
+      setSecurityResults(queryResponse || [])
+      setIsSearchLoading(false)
+    }, 1000)
+  }
+
+  const onDropDownSecurityClick = security => {
+    setSecurityResults([])
+
+    dispatch(tradeActions.selectSecurity(security))
+
+    history.push('/trade')
+  }
```

Comment:
So does this automatically put the result from the action into setSecurityResults ?

Reply:
Sorta.
Line 88 - Wipe our local state so the dropdown items go away
Line 90 - Dispatch the selectSecurity action with the security info of the option we clicked on. This in turn goes to our saga, fetches all the data, and eventually populates it in our store.
Line 92 - Just redirects the user to the trade page